components/credentials/credentials-controller.js

```diff
@@ -20,6 +33,58 @@ function factory($scope, config, $location, ipCookie, localStorageService) {
   var trustedDomainPrefix = 'settings.trustedDomains.';
   var trustedDomainKey = trustedDomainPrefix + self.callbackHostName;
 
+  if(
+    self.identity.id !== self.requestOwner &&
+    self.identity.credential[0]['@graph'].claim.id !== self.requestOwner) {
+    brAlertService.add(
+      'error', 'Identity document does not correspond with the request.');
+    throw new Error('Identity document does not correspond with the request.');
+  }
+  var session = null;
+  var sessionKey = [self.identity.id.replace(/[:]/g,''), 'session'].join('.');
+  if(requestData.keyType === keyTypes.EXISTING) {
+    session = ipCookie(sessionKey);
+    if(session === undefined) {
+      // NOTE: stripping colon because browser encodes it
+      // this could be an acknowledgement where the owner is in claim.id
+      sessionKey = [self.identity.credential[0]['@graph']
+        .claim.id.replace(/[:]/g,''), 'session'].join('.');
+      session = ipCookie(sessionKey);
+    }
+    if(session === undefined) {
+      // the information in the document does not correspond with an identity
+      brAlertService.add('error', 'Active session not found.');
+      throw new Error('Error: Active session not found.');
+    }
+    // refresh session
+    ipCookie(sessionKey, session, {
+      expires: 120,
+      expirationUnit: 'minutes',
+      secure: true
+    });
+  }
+  if(requestData.keyType !== keyTypes.EXISTING) {
+    console.log('IN TEMPORARY', sessionKey);
+    try {
+      session = JSON.parse(sessionStorage.getItem(sessionKey));
+      if(!session) {
+        // this could be an acknowledgement where the owner is in claim.id
+        sessionKey = [self.identity.credential[0]['@graph']
+          .claim.id.replace(/[:]/g,''), 'session'].join('.');
```

Comment:
Creating a function to get a sessionKey from an identity doc would be a good idea I think. Looks like this is done multiple times and could be error prone.